service/history/workflowExecutionContext.go

```diff
@@ -192,6 +199,43 @@ func (c *workflowExecutionContext) updateWorkflowExecution(transferTasks []persi
 	// Update went through so update the condition for new updates
 	c.updateCondition = c.msBuilder.GetNextEventID()
 	c.msBuilder.executionInfo.LastUpdatedTimestamp = time.Now()
+
+	// for all subscriber which watch changes of this workflow execution context
+	for _, channel := range c.subscriberNotificationChan {
+		select {
+		case channel <- true:
+			// this only notify watcher that there is a change
+		default:
+			// in case the channel is already filled with message
+			// this should NOT happen, unless there is a bug
+		}
+	}
+	return nil
+}
+
+func (c *workflowExecutionContext) watchWorkflowExecution() (string, chan bool, error) {
+	subscriberID := uuid.NewUUID().String()
+	if _, ok := c.subscriberNotificationChan[subscriberID]; ok {
+		// UUID collision
```

Comment:
UUID won't collide

Reply:
most cases UUID will not collide, 
but when that happen, wield bug will appear